### DISCUSSION TITLE
falcon.rb for sinatra

### DISCUSSION THREAD

**Discussion by marek22k** [socketry/falcon#207] [Q&A]
Hello,

I would like to deploy a Sinatra application with Falcon in a production environment. For development I have always used rackup, however the falcon website states that this is not recommended for production. However, `falcon host` requires a falcon.rb for production and I could not find an example for Sinatra.
The following is my config.ru:
```ruby
# frozen_string_literal: true
# sharable_constant_value: litera

require 'bundler'

Bundler.require

require 'rack/protection'
require 'rack-timeout'
require 'sinatra'
require_relative 'app/bgplookup'

use Rack::Timeout, service_timeout: 30
use Rack::Protection, except: :session_hijacking

require_relative 'lib/config/config'
$Config = Config.from_file('config.json')
$DB = $Config.db
$Logger = $Config.logger

require_relative 'lib/readers/helper'
$Readers = Readers.create_readers $DB

run BgpLookup
```
How can I create a falcon.rb from this?

**Comment by ioquatix** [comment]
I hope this can help you:

https://github.com/socketry/falcon-example-sinatra

Let me know if you have specific questions.

**Reply by marek22k** [reply]
Is there also an example of how I can port rack extensions like Rack::Timeout into the falcon.rb?
